File: BackEnd/TicketBooking.Domain/Entities/Organizer.cs
using TicketBooking.Domain.Common;
using TicketBooking.Domain.Enums;

namespace TicketBooking.Domain.Entities;

public class Organizer : AuditableEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? LogoUrl { get; set; }
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public OrganizerStatus Status { get; set; } = OrganizerStatus.PENDING;

    public ICollection<Event> Events { get; set; } = new List<Event>();
}

File: frontend/ticket-booking-web/src/domain/entities/Organizer.ts
import type { OrganizerStatus } from '../enums/OrganizerStatus';

export interface Organizer {
  id: string;
  userId: string;
  name: string;
  description?: string;
  website?: string;
  phone?: string;
  email?: string;
  status: OrganizerStatus;
  createdAt: string;
  updatedAt?: string;
}
